### PR TITLE
gh-86 Enable Graphs to be constructed from URLs.

### PR DESCRIPTION
**Pull request by GCHQDeveloper404**

### CODE REVIEW

File: core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/StreamUtil.java [hunk 1]
```diff
@@ -165,6 +168,43 @@ public static InputStream opAuths(final Class clazz, final boolean logErrors) {
         return schemas;
     }
 
+
+    public static InputStream[] openStream(final String... urls) throws IOException {
+        final InputStream[] rtn = new InputStream[urls.length];
+        int count = 0;
+        for (final String url : urls) {
+            rtn[count++] = openStream(url);
```
Comment (p013570) [inline core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/StreamUtil.java]: If this throws an exception then you need to ensure the rtn InputStreams are closed.

File: core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/StreamUtil.java [hunk 2]
```diff
@@ -165,6 +168,43 @@ public static InputStream opAuths(final Class clazz, final boolean logErrors) {
         return schemas;
     }
 
+
+    public static InputStream[] openStream(final String... urls) throws IOException {
```
Comment (p013570) [inline core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/StreamUtil.java]: Perhaps remove these methods and just have the methods that take a URL. This method could be confused with the open stream from path.

File: core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java [hunk 3]
```diff
@@ -358,6 +380,26 @@ public Builder addSchema(final InputStream schemaStream) {
             }
         }
 
+        public Builder addSchema(final URL schemaURL) {
```
Comment (p013570) [inline core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java]: I don't think we need these builder methods. Users can just use the StreamUtil class to convert a URL into an InputStream and then use the addSchema(final InputStream schema) method.

Comment (GCHQDeveloper404) [inline core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java]: Doesn't this just save the user a step, make for easier cleaner code?

File: core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/StreamUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        //Given
+        final URL resource = getClass().getClassLoader().getResource(FILE_NAME);
+        if (resource == null) fail("Test json file not found:" + FILE_NAME);
+
+        //When
+        final InputStream[] inputStreams = StreamUtil.openStreams(resource);
+
+        //Then
+        assertNotNull(inputStreams);
+        assertFalse("InputStreams length is 0", inputStreams.length == 0);
```
Comment (p013570) [inline core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/StreamUtilTest.java]: Just need to close the input streams in a finally block.

File: rest-api/core-rest/src/main/java/uk/gov/gchq/gaffer/rest/service/IOperationService.java [hunk 5]
```diff
@@ -45,10 +45,9 @@
 @Consumes(MediaType.APPLICATION_JSON)
 @Produces(MediaType.APPLICATION_JSON)
 public interface IOperationService {
-
     @POST
     @ApiOperation(value = "Performs the given operation chain on the graph", response = Object.class)
-    Object execute(final OperationChain opChain);
+    Object execute(final OperationChain opChain); //takes in operation TODO
```
Comment (p013570) [inline rest-api/core-rest/src/main/java/uk/gov/gchq/gaffer/rest/service/IOperationService.java]: Remove todo.

File: core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/StreamUtil.java [hunk 6]
```diff
@@ -128,7 +128,43 @@ public static InputStream authScores(final Class clazz) {
         return schemas;
     }
 
+    public static InputStream[] openStreams(final boolean isLogging, final URL... urls) throws IOException {
```
Comment (p013570) [inline core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/StreamUtil.java]: Don't think we need the isLogging parameter.

**Pull request merged**